### PR TITLE
Added encryption parameters validations

### PR DESCRIPTION
**Pull request by bhagyashree-sarawate**
Added encryption parameters validations as per the cases mentioned in PR#246

hostEncryption must be true/false any other value should throw error . 
hostEncryptionName/hostEncryptionNamespace both must be specified with a valid value for volume encryption
handled the scenarios when hostEncryption is not specified, true, false, invalid values 
@sneharai4 @raunakkumar @datamattsson , @rgcostea Request you to review

### CODE REVIEW

File: pkg/driver/controller_server.go [hunk 1]
```diff
@@ -1624,3 +1636,49 @@ func getSnapshotsForStorageProvider(ctx context.Context, request *csi.ListSnapsh
...
+
+	if hostEncryptionVal == "" && !encryptionSecretNameSpecified && !encryptionSecretNamespaceSpecified {
+		return falseKey, nil
+	}
+	
+	// If both keys are specified and no empty
+	if createParameters[hostEncryptionSecretNameKey] != "" && createParameters[hostEncryptionSecretNamespaceKey] != "" {
+		// when hostencryption is unspecified default value is empty
+		if hostEncryptionVal == "" {
+			return trueKey, nil
```
[the remark this answers is not in the document]
Comment (bhagyashree-sarawate) [reply pkg/driver/controller_server.go:1670]: to support the case1 mentioned in PR#246, setting hostEncryptionSecretName and hostEncryptionSecretNamespace will automatically qualify hostEncryption to "true" even though it is not specified ,

File: pkg/driver/controller_server.go [hunk 2]
```diff
@@ -346,6 +346,14 @@ func (driver *Driver) createVolume(
 		delete(createParameters, protectionTemplateKey)
 	}
 
+	// Validate the host encryption parameters 
+	hostEncryption, err := driver.validateHostEncryptionParameters(createParameters)  
+	if err != nil {
+		log.Error("err: ", err.Error())
+		return nil, status.Error(codes.InvalidArgument, fmt.Sprintf("Invalid arguments for encryption, %s", err.Error()))
+	} 
+ 	createParameters[hostEncryptionKey] = hostEncryption
```
[the remark this answers is not in the document]
Comment (bhagyashree-sarawate) [reply pkg/driver/controller_server.go:355]: hostEncryption can be true and false after validation  which can be used for the during Publish and Stage workflows

File: pkg/driver/controller_server.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (raunakkumar) [reply pkg/driver/controller_server.go:1670]: Could you elaborate more on the exact case? If we do treat is true then we don't need this flag.

[the remark this answers is not in the document]
Comment (bhagyashree-sarawate) [reply pkg/driver/controller_server.go:1667]: we have discussed this on mail

**Review by sneharai4 — APPROVED** [review]

**Pull request merged**